src/compiler/transformers/module/module.ts

```diff
@@ -545,14 +548,28 @@ namespace ts {
             //          : new Promise(function (_a, _b) { require([x], _a, _b); }); /*Amd Require*/
             // });
             needUMDDynamicImportHelper = true;
-            return createConditional(
-                /*condition*/ createIdentifier("__syncRequire"),
-                /*whenTrue*/ transformImportCallExpressionCommonJS(node),
-                /*whenFalse*/ transformImportCallExpressionAMD(node)
-            );
+            if (isSimpleCopiableExpression(arg)) {
+                const argClone = isGeneratedIdentifier(arg) ? arg : setEmitFlags(setTextRange(getSynthesizedClone(arg), arg), EmitFlags.NoComments);
+                if (argClone.kind === SyntaxKind.StringLiteral && typeof (argClone as StringLiteral).singleQuote === "undefined") {
+                    (argClone as StringLiteral).singleQuote = getSourceTextOfNodeFromSourceFile(currentSourceFile, arg).charCodeAt(0) === CharacterCodes.singleQuote;

```

Comment:
You can use `createLiteral(arg)` if `arg` is a string literal and it will create a fresh literal that gets its content from the provided literal.

Alternatively, we should consider just setting `singleQuote` when we parse a string literal...

Reply:
I want to keep using `getSynthesizedClone` since `undefined`, `null`, `Infinity`, and other identifiers/keywords are valid here.

We really should consider setting the singleQuote bit on all string literals...

Reply:
I understand that for non-string-literals, but as I said above: "you can use `createLiteral(arg)` if `arg` is a **string literal**". That will return a fresh string literal, but its source text is pulled from the original string literal when printed.

Basically:

```ts
const argClone = isGeneratedIdentifier(arg) ? arg :
    isStringLiteral(arg) ? createLiteral(arg) :
    setEmitFlags(setTextRange(getSynthesizedClone(arg), arg), EmitFlags.NoComments);
```

Reply:
Oh, instead of inspecting the content here! Understood, will do.